File: docman_flutter/lib/model/objects/Documento.dart
import 'Tag.dart';
import 'Utente.dart';

class Documento{
  int id;
  String titolo;
  String formato;
  DateTime data;
  String descrizione;
  int dimensione;
  String unitaDimensione;
  bool cestino;
  Utente proprietario;
  List<Tag> tags;
  List<Utente> condivisi;

  Documento({this.id, this.titolo, this.formato, this.data, this.descrizione, this.dimensione, this.unitaDimensione, this.cestino, this.proprietario, this.tags, this.condivisi});

  factory Documento.fromJson(Map<String, dynamic> json) {
    return Documento(
      id: json['id'],
      titolo: json['titolo'],
      formato: json['formato'],
      data: DateTime.parse(json['data']),
      descrizione: json['descrizione'],
      dimensione: json['dimensione'],
      unitaDimensione: json['unita_dimensione'],
      cestino: json['cestino'],
      proprietario: Utente.fromJson(json['proprietario']),
      tags: List<Tag>.from(json['tags'].map((i) => Tag.fromJson(i)).toList()),
      condivisi: List<Utente>.from(json['utenti'].map((i) => Utente.fromJson(i)).toList()),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'titolo': titolo,
    'formato': formato,
    'data': data,
    'descrizione': descrizione,
    'dimensione': dimensione,
    'unita_dimensione': unitaDimensione,
    'cestino': cestino,
    'proprietario': proprietario,
    'tags': tags,
    'utenti': condivisi,
  };
}

File: docman/src/main/java/it/ale/docman/entities/Documento.java
package it.ale.docman.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@Table(name = "documenti")
public class Documento {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "titolo", length = 100, nullable = false)
    private String titolo;

    @Column(name = "formato", length = 10, nullable = false)
    private String formato;

    @Column(name = "data", nullable = false)
    private LocalDateTime data;

    @Column(name = "descrizione", length = 250, nullable = false)
    private String descrizione;

    @Column(name = "dimensione", nullable = false)
    private int dimensione;

    @Column(name = "unita_dimensione", length = 2, nullable = false)
    private String unita_dimensione;

    @Column(name = "cestinoYN", nullable = false)
    private boolean cestino;

    @ManyToOne
    @JoinColumn(name = "proprietario")
    private Utente proprietario;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "associazioni_doc_tag", joinColumns = {@JoinColumn(name = "id_documento")}, inverseJoinColumns = {@JoinColumn(name = "id_tag")})
    private List<Tag> tags;

    @ManyToMany(mappedBy = "documentiCondivisi")
    private List<Utente> utenti;
}
